proxy-node-gateway/src/main/java/uk/gov/ida/notification/resources/HubResponseResource.java

```diff
@@ -19,72 +16,60 @@
+        @FormParam(SamlFormMessageType.SAML_RESPONSE) String hubResponse,
+        @FormParam("RelayState") String relayState,
+        @Session HttpSession session) {
 
-        try {
-            String expectedRequestId = (String) session.getAttribute("gateway_request_id");
-            hubResponseValidator.validate(encryptedHubResponse);
+        String hubRequestId = session.getAttribute(SESSION_KEY_HUB_REQUEST_ID).toString();
+        String eidasRequestId = session.getAttribute(SESSION_KEY_EIDAS_REQUEST_ID).toString();
+        String connectorEncrpytionCredential = session.getAttribute(SESSION_KEY_EIDAS_CONNECTOR_PUBLIC_CERT).toString();
 
-            if (!expectedRequestId.contains(encryptedHubResponse.getInResponseTo())) {
-                throw new InvalidHubResponseException(
-                    String.format("Received a Response from Hub for an AuthnRequest we have not seen (ID: %s)", encryptedHubResponse.getInResponseTo()));
-            }
+        LOG.info(String.format("[HUB Response] received for hub authn request ID '%s', eIDAS authn request ID '%s'", hubRequestId, eidasRequestId));
 
-            HubResponseContainer hubResponseContainer = HubResponseContainer.from(
-                    hubResponseValidator.getValidatedResponse(),
-                    hubResponseValidator.getValidatedAssertions()
-            );
+        HubResponseTranslatorRequest translatorRequest = new HubResponseTranslatorRequest(
+            hubResponse,
+            hubRequestId,
+            LEVEL_OF_ASSURANCE,
+            eidasRequestId,
+            connectorEncrpytionCredential
+        );
 
-            logHubResponse(hubResponseContainer);
-            Response eidasResponse = translatorService.getTranslatedResponse(encryptedHubResponse);
-            logEidasResponse(eidasResponse);
+        String eidasResponse = translatorProxy.getTranslatedResponse(translatorRequest);
 
-            return samlFormViewBuilder.buildResponse(
-                connectorNodeUrl,
-                eidasResponse,
-                "Post eIDAS Response SAML to Connector Node",
-                relayState
-            );
-        } catch (Throwable e) {
-            throw new HubResponseException(e, encryptedHubResponse);
-        }
-    }
+        LOG.info(String.format("[eIDAS Response] received for hub authn request ID '%s', eIDAS authn request ID '%s'", hubRequestId, eidasRequestId));
 
-    private void logHubResponse(HubResponseContainer hubResponseContainer) {
-        LOG.info("[Hub Response] ID: " + hubResponseContainer.getHubResponse().getResponseId());
-        LOG.info("[Hub Response] In response to: " + hubResponseContainer.getHubResponse().getInResponseTo());
-        LOG.info("[Hub Response] Provided level of assurance: " + hubResponseContainer.getAuthnAssertion().getProvidedLoa());
+        return samlFormViewBuilder.buildResponse(
+            connectorNodeUrl,
+            eidasResponse,
+            SUBMIT_TEXT,
+            relayState
```

Comment:
NB: In future we should return the `relayState` sent by the country in `AuthnRequest` in preference to the `relayState` send by the Hub. This will mean storing the country's `relayState` in session.

Reply:
Yeah that makes senes. I might as well change that as as part of this PR.